tests/nfs_pvc_retain_test.go

```diff
@@ -0,0 +1,273 @@
+			pvcObj, err := Client.getPVC(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while fetching pvc %s/%s", applicationNamespace, pvcName)
+			nfsPVName = pvcObj.Spec.VolumeName
+			// NOTE: Backend PVC name will be nfs-<nfs pv name>
+			backendPVCName = "nfs-" + nfsPVName
+
+			nfsServerLabelSelector := "openebs.io/nfs-server=nfs-" + pvcObj.Spec.VolumeName
+			err = Client.waitForPods(OpenEBSNamespace, nfsServerLabelSelector, corev1.PodRunning, 1)
+			Expect(err).To(BeNil(), "while verifying NFS Server running status")
+		})
+
+		It("should have sent details to server... verifying annotation on backend PVC", func() {
+			var backendPVCObj *corev1.PersistentVolumeClaim
+			var err error
+			var isEventReceived bool
+
+			for retries := 0; retries < maxRetryCount; retries++ {
+				backendPVCObj, err = Client.getPVC(OpenEBSNamespace, backendPVCName)
+				Expect(err).To(BeNil(), "while fetching backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+
+				_, isEventReceived = backendPVCObj.Annotations[nfs.VolumeCreateNFSPVCKey]
+				if isEventReceived {
+					break
+				}
+				// Reconciliation will happen at every 60 seconds but if any error occurs it will
+				// get reconcile easily
+				time.Sleep(time.Second * 10)
+			}
+			Expect(isEventReceived).To(BeTrue(), "NFS pvc %s/%s details are not exported to server", applicationNamespace, pvcName)
+			backendPVC, err := Client.getPVC(OpenEBSNamespace, backendPVCName)
+			Expect(err).To(BeNil(), "while fetching backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+			backendPVName = backendPVC.Spec.VolumeName
+
+			Expect(backendPVCObj.Annotations[nfs.VolumeCreateNFSPVCKey]).To(Equal(applicationNamespace+"-"+pvcName), "while verifying nfs pvc create event data")
+			Expect(backendPVCObj.Annotations[nfs.VolumeCreateNFSPVKey]).To(Equal(nfsPVName), "while verifying nfs pv create event data")
+			Expect(backendPVCObj.Annotations[nfs.VolumeCreateBackendPVCKey]).To(Equal(OpenEBSNamespace+"-"+backendPVCName), "while verifying backend pvc create event data")
+			Expect(backendPVCObj.Annotations[nfs.VolumeCreateBackendPVKey]).To(Equal(backendPVName), "while verifying backend pv create event data")
+		})
+	})
+
+	When("pvc "+pvcName+" with retain policy is deleted", func() {
+		It("should delete pvc and pv should remain in same state", func() {
+			err := Client.deletePVC(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while deleting pvc %s/%s", applicationNamespace, pvcName)
+
+			for retry := 5; retry >= 0; retry-- {
+				nfsPVObj, err := Client.getPV(nfsPVName)
+				Expect(err).To(BeNil(), "while fetching backend pv %s", nfsPVName)
+
+				Expect(nfsPVObj.DeletionTimestamp).To(BeNil(), "NFS pv %s shouldn't be marked for deletion", nfsPVName)
+				time.Sleep(time.Second * 5)
+			}
+		})
+		It("should not send events to server", func() {
+			var isDeleteEventReceived bool
+			for retry := 5; retry >= 0; retry-- {
+				backendPVCObj, err := Client.getPVC(OpenEBSNamespace, backendPVCName)
```

Comment:
nit: better to verification for non-empty `backendPVCName`. also `nfsPVName` and `backendPVName` wherever required.

Reply:
Done